packages/grafana-prometheus/src/querybuilder/components/MetricCombobox.tsx

```diff
@@ -0,0 +1,125 @@
+import { useCallback } from 'react';
+
+import { SelectableValue } from '@grafana/data';
+import { EditorField, EditorFieldGroup } from '@grafana/experimental';
+import { InlineField, InlineFieldRow } from '@grafana/ui';
+import { Combobox, ComboboxOption } from '@grafana/ui/src/components/Combobox/Combobox';
```

Comment:
Looking good! It feels interesting to see this without all the extra code for the metrics explorer, the truncated list and the footer message. Nice!

First note: we will need to get this exported with '@grafana/ui` before we make this official cause of the package and using it in AMP. But, for testing only in cloud, this works for now.

Reply:
Since the package is being built based on Grafana builds, I believe this PR is blocked by this import 😞. While we could ask @grafana/aws-datasources to pause their update of the AMP data source, and not use the newest version, this blocks other Prometheus upgrades that have been asked for by Amazon (e.g. introducing limit). I would prefer not to have to add admin tasks for them to keep an eye on the work around the combobox.

Is there anyway to conditionally import this or not include this in the build? @jackw interested in your thoughts. This seems kind of extra though, and could be solved by having the combobox exported so it could be used outside Grafana. What's the ETA on that work?

```
import { Combobox, ComboboxOption } from '@grafana/ui/src/components/Combobox/Combobox';
// or
import { Combobox, ComboboxOption } from '@grafana/ui';
```

Reply:
@bohandley This now uses the main import.

Reply:
>Is there anyway to conditionally import this or not include this in the build? @jackw interested in your thoughts. This seems kind of extra though, and could be solved by having the combobox exported so it could be used outside Grafana. What's the ETA on that work?

Apologises for not seeing this sooner @bohandley . You cannot access nested directories within grafana/ui from within a plugin. SystemJS doesn't work like that.

If you would like to expose the unstable entrypoint from grafana/ui, this could be done in a way that plugins could access it like:  `import { ComboBox } from "@grafana/ui/unstable"`. It wouldn't take much effort to achieve this with modifications to the `sharedDependencies.ts` and to the `@grafana/ui` build configs so plugin devs gain access to the unstable goodies.

Maybe we can take this discussion further somewhere on slack as it appears the solution here was to promote the combobox to stable?

Reply:
> You cannot access nested directories within grafana/ui from within a plugin. SystemJS doesn't work like that.

@jackw it is all good! This is what I wanted to know, thank you! The import issue was solved for the Prometheus package. There is a slack thread where your suggestion what mentioned with importing from `@grafana/ui/unstable` [here](https://raintank-corp.slack.com/archives/C025WTVRBSN/p1730285644404989). This might be something @grafana/frontend-plugins folks might want to look into in the future, it sounds sensible to me 🙂